src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorCodeDocumentExtensions.cs

```diff
@@ -303,12 +303,15 @@ public static bool TryComputeClassName(this RazorCodeDocument codeDocument, [Not
     }
 #nullable disable
 
+    public static bool TryComputeNamespace(this RazorCodeDocument document, bool fallbackToRootNamespace, out string @namespace)
+        => TryComputeNamespace(document, fallbackToRootNamespace: fallbackToRootNamespace, allowEmptyRootNamespace: false, out @namespace);
```

Comment:
MVC scenarios still call this overload. They already kinda work - you can see that in the added test - they use a constant fallback namespace name, but do not append hash to class name, so users could be referencing them. Razor components in global namespace, on the other hand, used also fallback class name with appended hash so it would be difficult for users to reference them.